### PR TITLE
[DOC] Tracing: Add doc for Tempo Search query

### PR DESCRIPTION
**Pull request by knylander-grafana**
**What is this feature?**

Adds documentation for the Tempo data source Search query type.  (see https://github.com/grafana/grafana/pull/64616)

The content for the query editor will be shared between Grafana, Grafana Cloud, and possibly Tempo. The main content has been added in the shared directory and then is included in the Tempo query editor markdown file. 

**Who is this feature for?**

Tracing users in Grafana who want to use the updated Search as a preview. 

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by knylander-grafana** [comment]
@jdbaldry FYI. This is the PR that sets up sharing an include file from Grafana to Grafana Cloud content.

File: docs/sources/datasources/tempo/query-editor/index.md
Comment (jdbaldry) [inline docs/sources/datasources/tempo/query-editor/index.md:76]: A note about HTML comments. These do end up in the resulting HTML page source.

You can instead write a "Markdown comment" with:

```suggestion
[//]: # (Include content for preview of Search tab featuring TraceQL query builder)
```

File: docs/sources/datasources/tempo/query-editor/index.md
Comment (jdbaldry) [inline docs/sources/datasources/tempo/query-editor/index.md:78]: Should the version be `"latest"` or should it be omitted? If it is omitted, the shortcode uses the current version. If you keep `"latest"`, it will use the shared document from "latest" in all versions that this shortcode exists in.

Note that until 10.0.x is released, or this is backported to 9.5.x, the shared page won't actually exist in "latest".

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (jdbaldry) [inline docs/sources/shared/datasources/tempo-search-traceql.md:14]: Nice! Really helpful comment.

Another place where we can consider using "Markdown comments" instead of HTML comments to avoid this ending up the page HTML.

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (jdbaldry) [inline docs/sources/shared/datasources/tempo-search-traceql.md:18]:
```suggestion
{{% admonition type="note" %}}
```

Admonition types should be specified in lower case.

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (jdbaldry) [inline docs/sources/shared/datasources/tempo-search-traceql.md:19]: I'm not sure about the parenthetical here, it doesn't seem that helpful to a reader.

**Review by jdbaldry — COMMENTED** [review]
I've not done a full review of the prose, I'll leave that to another technical writer but I've added some comments for things that jumped out whilst scanning.

**Comment by jdbaldry** [comment]
You'll need to run `prettier -w docs/sources`, double check the pages with `make docs`, and commit the changes to please the linter.

File: docs/sources/datasources/tempo/query-editor/index.md
Comment (knylander-grafana) [reply docs/sources/datasources/tempo/query-editor/index.md:76]: Thank you, @jdbaldry. I didn't know how to do comments in the Markdown files for Hugo besides using the HTML comments.

File: docs/sources/datasources/tempo/query-editor/index.md
Comment (knylander-grafana) [reply docs/sources/datasources/tempo/query-editor/index.md:78]: Oh, good point. Then we should definitely omit latest.

Comment (knylander-grafana) [inline docs/sources/datasources/tempo/query-editor/index.md:78]:
```suggestion
{{< docs/shared source="grafana" lookup="datasources/tempo-search-traceql.md" leveloffset="+1" >}}
```

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [reply docs/sources/shared/datasources/tempo-search-traceql.md:18]: Thank you. I had some trouble with the admonition rendering and was playing with it to see if it made a difference. I built it today and things seem to be okay now.

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [reply docs/sources/shared/datasources/tempo-search-traceql.md:19]: Because the link is missing :)

Comment (knylander-grafana) [reply docs/sources/shared/datasources/tempo-search-traceql.md:19]:
```suggestion
This new Search query type is available as an experimental feature. This feature is under active development and is not production ready. To try this feature, enable the `traceqlSearch` feature flag in Grafana (read [documentation](/docs/grafana/latest/setup-grafana/configure-grafana/feature-toggles/)). Grafana Cloud users should contact Grafana Support to enable this feature.
```

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [inline docs/sources/shared/datasources/tempo-search-traceql.md:42]:
```suggestion
|3|Choose filter|Choose whether to filter using **Resource Service Name**, **Span Name**, and/or **Duration**.|You can execute an empty query in the Search tab. In TraceQL, `{}` is a valid query and is the default query until you choose options.|
```

Comment (knylander-grafana) [reply docs/sources/shared/datasources/tempo-search-traceql.md:42]: Thank you. I've updated the text to reflect this.

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [inline docs/sources/shared/datasources/tempo-search-traceql.md:14]:
```suggestion

[//]: # (This file documents the Search query type for the Tempo data source. It is available as a public preview.)

[//]: # (This shared file is included in these locations:)
[//]: # (/grafana/docs/sources/datasources/tempo/query-editor/index.md)

[//]: # (If you make changes to this file, verify that the meaning and content are not changed in any place where the file is included.)

[//]: # (Any links should be fully qualified and not relative: /docs/grafana/) 
```

File: docs/sources/datasources/tempo/query-editor/index.md
Comment (knylander-grafana) [reply docs/sources/datasources/tempo/query-editor/index.md]: Done.

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [reply docs/sources/shared/datasources/tempo-search-traceql.md]: Thank you. I've updated the comments to use the Markdown comments you highlighted earlier.

**Comment by knylander-grafana** [comment]
> You'll need to run `prettier -w docs/sources`, double check the pages with `make docs`, and commit the changes to please the linter.

Thank you for the specific command. I actually ran prettier last night and commited the changes, but I didn't use the prettier -w command you provided.

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [inline docs/sources/shared/datasources/tempo-search-traceql.md:60]:
```suggestion
**Duration** is a duration field type and uses range selections (`>`, `>=`, `<`, `<=`). When you select multiple values for the same tag, Grafana automatically changes the operator to the regex operator `=~` and concatenates the values with a `|`.
```

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [reply docs/sources/shared/datasources/tempo-search-traceql.md:68]: Thank you for this information. I've added the content to line 60 above, so it's in the paragraph with duration.

**Comment by knylander-grafana** [comment]
@adrapereira Thank you for your review. I've updated the content with your suggestions. Would you please re-review?

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [inline docs/sources/shared/datasources/tempo-search-traceql.md:61]:
```suggestion
When you select multiple values for the same tag, Grafana automatically changes the operator to the regex operator `=~` and concatenates the values with a `|`.

**Duration** is a duration field type and uses range selections (`>`, `>=`, `<`, `<=`). 
```

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [reply docs/sources/shared/datasources/tempo-search-traceql.md:69]: Thank you for the great info! I've added another paragraph. What do you think?

File: docs/sources/shared/datasources/tempo-search-traceql.md
Comment (knylander-grafana) [reply docs/sources/shared/datasources/tempo-search-traceql.md:61]: Line 67 should be in a code block. The formatting in the suggestion box doesn't like the additional code block. I'll fix it once we get the suggestion wording right.

Comment (knylander-grafana) [reply docs/sources/shared/datasources/tempo-search-traceql.md:61]: @adrapereira How does this update look?

**Review by adrapereira — APPROVED** [review]
Thank you for the updates!

**Comment by knylander-grafana** [comment]
@imatwawana Can this be included in the G10 docs?

**Comment by jdbaldry** [comment]
I've updated the milestone to `10.1.x` as that is the version targeted by the `main` branch.

Once this PR is merged and backported to the `v10.0.x` version branch, your changes will be part of the documentation for Grafana 10.

**Review by chri2547 — APPROVED** [review]

**Pull request merged**
